mdc_weight_mgmt_xml_import/models/maintenance_equipment.py

```diff
@@ -0,0 +1,156 @@
+# # © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# # License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import models, api, _, fields
+from datetime import datetime
+from lxml import etree as ET
+from pathlib import Path
+import traceback
+import logging
+import re
+
+
+class MaintenanceEquipment(models.Model):
+    _inherit = 'maintenance.equipment'
+
+    mdc_xml_import_active = fields.Boolean(
+        string="XML File Import Active"
+    )
+    mdc_xml_weight_in_path = fields.Char(
+        string="XML Input Folder Path",
+        help="Path to the folder where unprocessed XML files are placed."
+    )
+    mdc_xml_weight_processed_path = fields.Char(
+        string="XML Processed Folder Path",
+        help="Path to the folder where processed XML files are moved."
+    )
+
+    @api.model
+    def _cron_import_xml_weight(self):
+        logger = logging.getLogger(__name__)
+        errors = []
+        for record in self.env['maintenance.equipment'].search([
+                    ('mdc_xml_import_active', '=', True)
+                ]):
+            in_path = Path(record.mdc_xml_weight_in_path)
+            processed_path = Path(record.mdc_xml_weight_processed_path)
+
+            try:
+                in_path.mkdir(parents=True, exist_ok=True)
+                processed_path.mkdir(parents=True, exist_ok=True)
+            except Exception as e:
+                logger.error("Can't create folders: %s", e)
+                errors.append({
+                    'company_id': record.company_id.id,
```

Comment:
A maintenance equipment could be created without company, as its global company rule shows:

https://github.com/odoo/odoo/blob/2c5fb1260c662adc7f48ecdbfdeba91f964d87bb/addons/maintenance/security/maintenance.xml#L49-L53

so we should protect this:
```suggestion
                    'company_id': record.company_id and record.company_id.id or False,
```